File: src/main/java/com/sasfmlzr/apivk/thread/ThreadUserBot.java
package com.sasfmlzr.apivk.thread;

import com.sasfmlzr.apivk.client.BotApiClient;
import com.sasfmlzr.apivk.object.StatisticsVariable;
import com.sasfmlzr.apivk.object.UserRights;
import com.vk.api.sdk.client.actors.UserActor;
import com.vk.api.sdk.exceptions.ApiException;
import com.vk.api.sdk.exceptions.ClientException;
import com.vk.api.sdk.objects.messages.Dialog;

import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

public class ThreadUserBot extends ThreadBot implements Runnable        //(содержащее метод run())          отправление сообщения в рекурсии в отдельном потоке
{
    private BotApiClient client;

    public ThreadUserBot(BotApiClient client, UserActor actor) {
        this.client = client;
        this.actor = actor;
    }

    private boolean stoped = false;
    private UserActor actor;

    public void run()         //Этот метод будет выполняться в побочном потоке
    {
        while (!stoped) {
            boolean exception = false;
            try {
                sendMessageUser(actor);
            } catch (ClientException | SQLException | InterruptedException | ApiException e) {
                exception = true;
                e.printStackTrace();
                System.out.print("Исключение в потоке бота \n");
            }
            if (!exception) {
                stopped();
            } else {
                try {
                    Thread.sleep(1000);
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }
            }
        }
        System.out.print("Побочный поток завершён \n");
    }

    private void stopped() {
        stoped = true;
    }

    //-----------------отправка сообщения, если есть непрочитанные-----------------//
    private void sendMessageUser(UserActor actor) throws ClientException, ApiException, InterruptedException, SQLException {

        while (client.getStateBot().getPushPowerBot()) {
            client.getStateBot().setFindMessage(false);        // совпадение с сообщением не найдено
            StatisticsVariable.countSendMessageUser = StatisticsVariable.countSendMessageUser + 1;
            long timeStartFunction = System.currentTimeMillis();


            client.getStateBot().setBotWork(true);           // если метод запущен, то бот включен
            client.getStateBot().setPriostanovka(false);   // для приостановки бота
            String message;         // сообщение бота
            String obrachenie = "Колян, ";                      //обращение к боту
            message = client.Companion.getDatabase().getBotRandomData().
                    get(client.other().randomId(client.Companion.getDatabase().
                            getBotRandomData().size())).response;          //сообщение берется из рандомной базы коляна
                        /*
           if (textMessageString.equals("")){
               System.out.print("textMessageString = " +textMessageString+ "\n");
               message = "Я Бот Колян, я выпил Блэйзера и не могу отвечать на пустые сообщения(в которых только вложения)";
           }*/
            List<Dialog> messagesList = createListMessageVK();                   // делается запрос непрочитанных сообщений
            obrachenie = "Колян, ";
            StatisticsVariable.timeZaprosFinishSumm = StatisticsVariable.timeZaprosFinishSumm + StatisticsVariable.timeZaprosFinishItogo;            // для среднего пинга


            UserRights userRight;

            if (messagesList.size() != 0) {
                int userID = messagesList.get(0).getMessage().getUserId();               // запись userID пользователя
                client.Companion.getDatabase().databaseRequest(client.Companion.getDatabase().getStatmt()).addInfoUser(userID, actor, client.getVkApiClient());       // добавить инфу о пользователе, если нет  // здесь есть запрос к вк
                client.Companion.getDatabase().databaseRequest(client.Companion.getDatabase().getStatmt()).addInfoUserRights(userID, actor);    // добавить права пользователю, если нет
                String userRightString = client.Companion.getDatabase().databaseRequest(client.Companion.getDatabase().getStatmt()).findUserRights(userID, actor);  // запись права текущего пользователя в ячейку
                userRight = new UserRights(userRightString);  // наследование прав пользователем
                //    System.out.print(userRight.getNameRight() + " может админить? -" + userRight.getAdminCommands() + "\n" +
                //           userRight.getNameRight() + " может писать боту? -" + userRight.getAllowWriteToBot() + "\n");


                String textMessageString = messagesList.get(0).getMessage().getBody().toLowerCase();       // прием сообщения в переменную
                if (userRight.getAdminCommands()) {
                    if (!client.getStateBot().getBotStopped()) {     // если совпадение с сообщением не найдено, то
                        if (textMessageString.contains(obrachenie.toLowerCase())) {

                            textMessageString = textMessageString.replaceAll(obrachenie.toLowerCase(), "");
                            textMessageString = textMessageString.replaceAll("[^ A-Za-zА-Яа-я0-9?]", "");       // замена знаков
                            String dublicateMessages = message;
                            message = client.messages().commands().adminCommandsBot(textMessageString, message);         //проверка на команды бота
                            if (!Objects.equals(message, dublicateMessages)) {
                                client.messages().vkSendMessage(actor, message, messagesList);
                                messagesList.clear();
                            }
                        }
                    }
                }
            }

            if (!client.getStateBot().getBotStopped()) {

                if (messagesList.size() != 0) {                                             // если нет непрочитанных, сообщений, то не выполнять
                    String textMessageString = messagesList.get(0).getMessage().getBody().toLowerCase();       // прием сообщения в переменную              TEST


                    //    statmt.execute("SELECT 'login','userID','UserRights' FROM 'UserRights' WHERE login="+actor.getId()+" AND userID="+userID);
                    client.getStateBot().setReduction(false);
                    if (!client.getStateBot().getFindMessage()) {     // если совпадение с сообщением не найдено, то
                        if (textMessageString.contains(obrachenie.toLowerCase())) {
                            textMessageString = textMessageString.replaceAll(obrachenie.toLowerCase(), "");
                            message = client.messages().commands().commandsBot(textMessageString, message, actor, messagesList, client);         //проверка на команды бота
                            textMessageString = textMessageString.replaceAll("[^ A-Za-zА-Яа-я0-9?]", "");       // замена знаков
                        }
                    }
                    if (!client.getStateBot().getReduction()) {
                        message = messageFromDataBase(textMessageString, message);       // бд коляна
                        message = messageFromBigDataBase(textMessageString, message);    // большая бд
                        client.messages().vkSendMessage(actor, message, messagesList);       // отправка сообщения и пометка его прочитанным
                    } else {
                        Thread.sleep(500);
                        //            System.out.print("Нет новых сообщений" + "\n");
                    }


                    long timeFinishFunction = System.currentTimeMillis();
                    long timeConsumedMilliss = timeFinishFunction - timeStartFunction;
                    //            System.out.print("время countSendMessageUser= " + timeConsumedMilliss + "\n");
                    StatisticsVariable.timeItogoMsMinusVK = timeConsumedMilliss - StatisticsVariable.timeZaprosFinishItogo;
                    //            System.out.print("время прохода минус запросвк= " + timeItogoMsMinusVK + "\n");
                }
                if (client.getStateBot().getPriostanovka()) {
                    //                System.out.print("бот приостановлен \n");
                    Thread.sleep(60000);
                }
                delayThread(messagesList, client);          // поток засыпает

            } else {
                Thread.sleep(2000);
            }

/////////////////////////статистика
            //StatisticsWindowController.seriesZaprosVk.getData().add(new XYChart.Data(countSendMessageUser, timeZaprosFinishItogo));          //ведение статистики запросов
            //StatisticsWindowController.seriesItogVk.getData().add(new XYChart.Data(countSendMessageUser, timeItogoMsMinusVK));          //ведение статистики запросов
            //StatisticsWindowController.seriesThread.getData().add(new XYChart.Data(countSendMessageUser, timeDelayThread));                        //ведение статистики задержки потока////здесь иногда ловится исключение
        }
        client.Companion.getDatabase().CloseDB();         //закрытие бд
        client.getStateBot().setBotWork(false);
    }

    //-----------------отправка и отслеживание запроса в вк на непрочитанные сообщения-------------------------------//
    private List<Dialog> createListMessageVK() throws ClientException, ApiException {
        long timezaprosstart = System.currentTimeMillis();         // начало запроса непрочитанного запроса
        List<Dialog> messages = client.getVkApiClient().messages().getDialogs(actor)                 // Листы сообщений
                .unread(true)
                .execute().getItems();
        long timezaprosfinish = System.currentTimeMillis();
        StatisticsVariable.timeZaprosFinishItogo = timezaprosfinish - timezaprosstart;      // время, затраченное на операцию
        return messages;
    }

    //-----------------поиск сообщения в основной БД----------------------------------------------//
    private String messageFromDataBase(String textMessageString, String message) {           //РАБОТА С ОСНОВНОЙ ТАБЛИЦЕЙ КОЛЯНА
        String messages = message;
        List<String> listMessages = new ArrayList<>();
        if (!client.getStateBot().getFindMessage()) {      // если совпадение с сообщением не найдено, то
            long timeStartBD = System.currentTimeMillis();
            // путешествие по списку объектов из БД
            for (int countDB = 0; countDB <= client.botApiClient().Companion.getDatabase().getBotData().size() - 1; countDB = countDB + 1) {
                if (client.Companion.getDatabase().getBotData().get(countDB).request.toLowerCase().equals(textMessageString.toLowerCase())) {  // сравниваем нижний регистр
                    listMessages.add(client.Companion.getDatabase().getBotData().get(countDB).response);
                    client.getStateBot().setFindMessage(true);                                                           // совпадение с сообщением найдено
                }
            }
            if (client.getStateBot().getFindMessage())    // если совпадение с сообщением найдено, то
                messages = listMessages.get(client.other().randomId(listMessages.size()));    // выбираем рандомно из найденного сообщение
            long timeFinishBD = System.currentTimeMillis();
            StatisticsVariable.timeConsumedMillisBD = timeFinishBD - timeStartBD;
            StatisticsVariable.countUsedBD = StatisticsVariable.countUsedBD + 1;                                              // количество использований бд коляна увеличилось на 1
            //        System.out.print("время операции по бд коляна= "+ timeConsumedMillisBD + "\n");
        }
        return messages;
    }

    //-----------------поиск сообщения в большой БД-----------------------------------------------//
    private String messageFromBigDataBase(String textMessageString, String message) {       //РАБОТА С ОСНОВНОЙ ТАБЛИЦЕЙ КОЛЯНА
        String messages = message;
        List<String> listMessages = new ArrayList<>();
        if (!client.getStateBot().getFindMessage()) {      // если совпадение с сообщением не найдено, то
            long timeStartBigBD = System.currentTimeMillis();
            // путешествие по списку объектов из большой БД

            /*
            resSettingBigBD = statmt.executeQuery("SELECT requesttextbot, responsetextbot FROM RandomBazaBot");
                        System.out.print("BigMessagesData= "+ BigMessagesData.size()+ "\n");
            while(resSettingBigBD.next())
            {
                if ( resSettingBigBD.getString("requesttextbot").equals(textMessageString))  {  // сравниваем нижний регистр
                    // сравниваем сообщение и значение в БД
                    listMessages.add(resSettingBigBD.getString("responsetextbot"));
                    findMessage=true;
                }
            }
            resSettingBigBD.close();
            if (findMessage)
                messages = listMessages.get(randomIdBot(listMessages.size()));    // выбираем рандомно из найденного сообщение
*/

            for (int countDB = 0; countDB <= client.Companion.getDatabase().getBigMessagesData().size() - 1; countDB = countDB + 1) {
                if (client.Companion.getDatabase().getBigMessagesData().get(countDB).request.toLowerCase().equals(textMessageString)) {  // сравниваем нижний регистр
                    // сравниваем сообщение и значение в БД
                    listMessages.add(client.Companion.getDatabase().getBigMessagesData().get(countDB).response);
                    client.getStateBot().setFindMessage(true);                                                 // совпадение с сообщением найдено
                    // messages = listMessages.get(randomIdBot(listMessages.size()));    // выбираем рандомно из найденного сообщение
                }
            }
            if (client.getStateBot().getFindMessage())                                                            // если совпадение с сообщением найдено, то
            {
                messages = listMessages.get(client.other().randomId(listMessages.size()));          // выбираем рандомно из найденного сообщение
            }
            long timeFinishBigBD = System.currentTimeMillis();
            StatisticsVariable.timeConsumedMillisBigBD = timeFinishBigBD - timeStartBigBD;
            StatisticsVariable.countUsedBigBD = StatisticsVariable.countUsedBigBD + 1;                                            // количество использований большой бд увеличилось на 1
            //StatisticsWindowController.seriesBigBD.getData().add(new XYChart.Data(countUsedBigBD, timeConsumedMillisBigBD));                        //ведение статистики задержки потока////здесь иногда ловится исключение
        }
        return messages;
    }
}

File: src/main/java/com/sasfmlzr/apivk/thread/ThreadUserBot.kt
package com.sasfmlzr.apivk.thread

import com.database.DatabaseEntity
import com.sasfmlzr.apivk.`object`.StatisticsVariable
import com.sasfmlzr.apivk.`object`.StatisticsVariable.countSendMessageUser
import com.sasfmlzr.apivk.`object`.StatisticsVariable.countUsedBigBD
import com.sasfmlzr.apivk.`object`.StatisticsVariable.timeConsumedMillisBigBD
import com.sasfmlzr.apivk.`object`.StatisticsVariable.timeDelayThread
import com.sasfmlzr.apivk.`object`.StatisticsVariable.timeItogoMsMinusVK
import com.sasfmlzr.apivk.`object`.StatisticsVariable.timeZaprosFinishItogo
import com.sasfmlzr.apivk.`object`.UserRights
import com.sasfmlzr.apivk.client.BotApiClient
import com.sasfmlzr.vkbot.controller.menuprogram.StatisticsWindowController
import com.vk.api.sdk.client.actors.UserActor
import com.vk.api.sdk.exceptions.ApiException
import com.vk.api.sdk.exceptions.ClientException
import com.vk.api.sdk.objects.enums.MessagesFilter
import com.vk.api.sdk.objects.messages.ConversationWithMessage
import javafx.scene.chart.XYChart
import java.sql.SQLException
import java.util.*

class ThreadUserBot(private val client: BotApiClient, private val actor: UserActor) : ThreadBot(),
        Runnable        //(содержащее метод run())          отправление сообщения в рекурсии в отдельном потоке
{

    private var stoped = false

    override fun run()         //Этот метод будет выполняться в побочном потоке
    {
        while (!stoped) {
            var exception = false
            try {
                sendMessageUser(actor)
            } catch (e: ClientException) {
                exception = true
                e.printStackTrace()
                print("Исключение в потоке бота \n")
            } catch (e: SQLException) {
                exception = true
                e.printStackTrace()
                print("Исключение в потоке бота \n")
            } catch (e: InterruptedException) {
                exception = true
                e.printStackTrace()
                print("Исключение в потоке бота \n")
            } catch (e: ApiException) {
                exception = true
                e.printStackTrace()
                print("Исключение в потоке бота \n")
            }

            if (!exception) {
                stopped()
            } else {
                try {
                    Thread.sleep(1000)
                } catch (e: InterruptedException) {
                    e.printStackTrace()
                }

            }
        }
        print("Побочный поток завершён \n")
    }

    private fun stopped() {
        stoped = true
    }

    //-----------------отправка сообщения, если есть непрочитанные-----------------//
    @Throws(ClientException::class, ApiException::class, InterruptedException::class, SQLException::class)
    private fun sendMessageUser(actor: UserActor) {

        while (client.stateBot.pushPowerBot) {
            client.stateBot.findMessage = false        // совпадение с сообщением не найдено
            countSendMessageUser = countSendMessageUser + 1
            val timeStartFunction = System.currentTimeMillis()

            client.stateBot.botWork = true           // если метод запущен, то бот включен
            client.stateBot.priostanovka = false   // для приостановки бота
            var message: String?         // сообщение бота
            var obrachenie = "Колян, "                      //обращение к боту
            message = DatabaseEntity.database.botRandomData[client.other().randomId(DatabaseEntity.database.botRandomData.size)]
                    .response          //сообщение берется из рандомной базы коляна
            /*
           if (textMessageString.equals("")){
               System.out.print("textMessageString = " +textMessageString+ "\n");
               message = "Я Бот Колян, я выпил Блэйзера и не могу отвечать на пустые сообщения(в которых только вложения)";
           }*/
            val messagesList = createListMessageVK()                   // делается запрос непрочитанных сообщений
            obrachenie = "Колян, "
            StatisticsVariable.timeZaprosFinishSumm =
                    StatisticsVariable.timeZaprosFinishSumm + timeZaprosFinishItogo            // для среднего пинга

            val userRight: UserRights

            if (messagesList.size != 0) {
                val userID = messagesList[0].lastMessage.peerId!!               // запись userID пользователя
                DatabaseEntity.database.databaseRequest.addInfoUser(
                        userID,
                        actor,
                        client.vk
                )       // добавить инфу о пользователе, если нет  // здесь есть запрос к вк
                DatabaseEntity.database.databaseRequest
                        .addInfoUserRights(userID, actor)    // добавить права пользователю, если нет
                val userRightString =
                        DatabaseEntity.database.databaseRequest
                                .findUserRights(userID, actor)  // запись права текущего пользователя в ячейку
                userRight = UserRights(userRightString)  // наследование прав пользователем
                //    System.out.print(userRight.getNameRight() + " может админить? -" + userRight.getAdminCommands() + "\n" +
                //           userRight.getNameRight() + " может писать боту? -" + userRight.getAllowWriteToBot() + "\n");


                var textMessageString = messagesList[0].lastMessage.text.toLowerCase()       // прием сообщения в переменную
                if (userRight.adminCommands!!) {
                    if (!client.stateBot.botStopped) {     // если совпадение с сообщением не найдено, то
                        if (textMessageString.contains(obrachenie.toLowerCase())) {

                            textMessageString = textMessageString.replace(obrachenie.toLowerCase().toRegex(), "")
                            textMessageString =
                                    textMessageString.replace("[^ A-Za-zА-Яа-я0-9?]".toRegex(), "")       // замена знаков
                            message = client.messages().commands()
                                    .adminCommandsBot(textMessageString, message)         //проверка на команды бота
                            if (message != message) {
                                client.messages().vkSendMessage(actor, message, messagesList)
                                messagesList.clear()
                            }
                        }
                    }
                }
            }

            if (!client.stateBot.botStopped) {

                if (messagesList.size != 0) {                                             // если нет непрочитанных, сообщений, то не выполнять
                    var textMessageString =
                            messagesList[0].lastMessage.text.toLowerCase()       // прием сообщения в переменную              TEST

                    //    statmt.execute("SELECT 'login','userID','UserRights' FROM 'UserRights' WHERE login="+actor.getId()+" AND userID="+userID);
                    client.stateBot.reduction = false
                    if (!client.stateBot.findMessage) {     // если совпадение с сообщением не найдено, то
                        if (textMessageString.contains(obrachenie.toLowerCase())) {
                            textMessageString = textMessageString.replace(obrachenie.toLowerCase().toRegex(), "")
                            message = client.messages().commands().commandsBot(
                                    textMessageString,
                                    message,
                                    actor,
                                    messagesList,
                                    client
                            )         //проверка на команды бота
                            textMessageString =
                                    textMessageString.replace("[^ A-Za-zА-Яа-я0-9?]".toRegex(), "")       // замена знаков
                        }
                    }
                    if (!client.stateBot.reduction) {
                        message = messageFromDataBase(textMessageString, message)       // бд коляна
                        message = messageFromBigDataBase(textMessageString, message!!)    // большая бд

                        if(textMessageString.contains("Красная") || textMessageString.contains("Белый") ||
                                textMessageString.contains("Красный") || textMessageString.contains("Белая")||
                                textMessageString.contains("Красн") || textMessageString.contains("Бел")||
                                textMessageString.contains("красн") || textMessageString.contains("бел")) {
                            message = textMessageString.toUpperCase() + "! " + textMessageString.toUpperCase() + "! "+ textMessageString.toUpperCase() + "!"
                        }

                        client.messages().vkSendMessage(
                                actor,
                                message,
                                messagesList
                        )       // отправка сообщения и пометка его прочитанным
                    } else {
                        Thread.sleep(500)
                        //            System.out.print("Нет новых сообщений" + "\n");
                    }


                    val timeFinishFunction = System.currentTimeMillis()
                    val timeConsumedMilliss = timeFinishFunction - timeStartFunction
                    //            System.out.print("время countSendMessageUser= " + timeConsumedMilliss + "\n");
                    timeItogoMsMinusVK =
                            timeConsumedMilliss - timeZaprosFinishItogo
                    //            System.out.print("время прохода минус запросвк= " + timeItogoMsMinusVK + "\n");
                }
                if (client.stateBot.priostanovka) {
                    //                System.out.print("бот приостановлен \n");
                    Thread.sleep(60000)
                }
                delayThread(messagesList, client)          // поток засыпает

            } else {
                Thread.sleep(2000)
            }

            /////////////////////////статистика
            StatisticsWindowController.seriesZaprosVk.data.add(XYChart.Data(countSendMessageUser, timeZaprosFinishItogo))          //ведение статистики запросов
            StatisticsWindowController.seriesItogVk.data.add(XYChart.Data(countSendMessageUser, timeItogoMsMinusVK))          //ведение статистики запросов
            StatisticsWindowController.seriesThread.data.add(XYChart.Data(countSendMessageUser, timeDelayThread))                        //ведение статистики задержки потока////здесь иногда ловится исключение
        }
        DatabaseEntity.database.CloseDB()         //закрытие бд
        client.stateBot.botWork = false
    }

    //-----------------отправка и отслеживание запроса в вк на непрочитанные сообщения-------------------------------//
    @Throws(ClientException::class, ApiException::class)
    private fun createListMessageVK(): MutableList<ConversationWithMessage> {
        val timezaprosstart = System.currentTimeMillis()         // начало запроса непрочитанного запроса
        val messages = client.vk.messages().getConversations(actor)                 // Листы сообщений
                .filter(MessagesFilter.UNREAD)
                .count(30)
                .execute().items
        val timezaprosfinish = System.currentTimeMillis()
        timeZaprosFinishItogo =
                timezaprosfinish - timezaprosstart      // время, затраченное на операцию
        return messages
    }

    //-----------------поиск сообщения в основной БД----------------------------------------------//
    private fun messageFromDataBase(
            textMessageString: String,
            message: String?
    ): String? {           //РАБОТА С ОСНОВНОЙ ТАБЛИЦЕЙ КОЛЯНА
        var messages: String? = message
        val listMessages = ArrayList<String>()
        if (!client.stateBot.findMessage) {      // если совпадение с сообщением не найдено, то
            val timeStartBD = System.currentTimeMillis()
            // путешествие по списку объектов из БД
            var countDB = 0
            while (countDB <= (DatabaseEntity.database.botData.size - 1)) {
                if (DatabaseEntity.database.botData[countDB].request.toLowerCase() == textMessageString.toLowerCase()
                ) {  // сравниваем нижний регистр
                    listMessages.add(DatabaseEntity.database.botData[countDB].response)
                    client.stateBot.findMessage =
                            true                                                           // совпадение с сообщением найдено
                }
                countDB += 1
            }
            if (client.stateBot.findMessage)
            // если совпадение с сообщением найдено, то
                messages =
                        listMessages[client.other().randomId(listMessages.size)]    // выбираем рандомно из найденного сообщение
            val timeFinishBD = System.currentTimeMillis()
            StatisticsVariable.timeConsumedMillisBD = timeFinishBD - timeStartBD
            StatisticsVariable.countUsedBD =
                    StatisticsVariable.countUsedBD + 1                                              // количество использований бд коляна увеличилось на 1
            //        System.out.print("время операции по бд коляна= "+ timeConsumedMillisBD + "\n");
        }
        return messages
    }

    //-----------------поиск сообщения в большой БД-----------------------------------------------//
    private fun messageFromBigDataBase(
            textMessageString: String,
            message: String
    ): String {       //РАБОТА С ОСНОВНОЙ ТАБЛИЦЕЙ КОЛЯНА
        var messages = message
        val listMessages = ArrayList<String>()
        if (!client.stateBot.findMessage) {      // если совпадение с сообщением не найдено, то
            val timeStartBigBD = System.currentTimeMillis()
            // путешествие по списку объектов из большой БД

            /*
            resSettingBigBD = statmt.executeQuery("SELECT requesttextbot, responsetextbot FROM RandomBazaBot");
                        System.out.print("BigMessagesData= "+ BigMessagesData.size()+ "\n");
            while(resSettingBigBD.next())
            {
                if ( resSettingBigBD.getString("requesttextbot").equals(textMessageString))  {  // сравниваем нижний регистр
                    // сравниваем сообщение и значение в БД
                    listMessages.add(resSettingBigBD.getString("responsetextbot"));
                    findMessage=true;
                }
            }
            resSettingBigBD.close();
            if (findMessage)
                messages = listMessages.get(randomIdBot(listMessages.size()));    // выбираем рандомно из найденного сообщение
*/

            var countDB = 0
            while (countDB <= DatabaseEntity.database.bigMessagesData.size - 1) {
                if (DatabaseEntity.database.bigMessagesData[countDB].request.toLowerCase() == textMessageString
                ) {  // сравниваем нижний регистр
                    // сравниваем сообщение и значение в БД
                    listMessages.add(DatabaseEntity.database.bigMessagesData[countDB].response)
                    client.stateBot.findMessage =
                            true                                                 // совпадение с сообщением найдено
                    // messages = listMessages.get(randomIdBot(listMessages.size()));    // выбираем рандомно из найденного сообщение
                }
                countDB += 1
            }
            if (client.stateBot.findMessage)
            // если совпадение с сообщением найдено, то
            {
                messages =
                        listMessages[client.other().randomId(listMessages.size)]          // выбираем рандомно из найденного сообщение
            }
            val timeFinishBigBD = System.currentTimeMillis()
            timeConsumedMillisBigBD = timeFinishBigBD - timeStartBigBD
            countUsedBigBD += 1                                            // количество использований большой бд увеличилось на 1
            StatisticsWindowController.seriesBigBD.data.add(XYChart.Data(countUsedBigBD, timeConsumedMillisBigBD.toInt()))                        //ведение статистики задержки потока////здесь иногда ловится исключение
        }
        return messages
    }
}
